src/ProjectDataProvider.ts

```diff
@@ -79,6 +81,62 @@ export module  ProjectDataProvider {
         });
     };
 
+    function isEmptyObject(obj) {
+        for(var prop in obj) {
+          if (Object.prototype.hasOwnProperty.call(obj, prop)) {
+            return false;
+          }
+        }
+        return true;
+    }
+
+    function clearEmptyObject(myObj) {
+        for(var key in myObj){
+            if(!(myObj[key] instanceof Array) && typeof(myObj[key]) == 'object' && isEmptyObject(myObj[key])){
+                delete myObj[key];
+            } 
+        }
+        return myObj;
+    }
+
+    function formatObj(myObj, keyArrays){
+        for(var key in myObj){
+            if(keyArrays.indexOf(key) == -1 && ( myObj[key] instanceof Array || typeof(myObj[key]) != 'object' || isEmptyObject(myObj[key]))){
+                delete myObj[key];
+            } else {
+                if(typeof(myObj[key]) == 'object') {
+                    formatObj(myObj[key],keyArrays);
+                }
+            }  
+        }
+        return myObj = clearEmptyObject(myObj);
+    }
+
+    formPackagedependencyNpmList = (item) => {
+        let manifestRootFolderPath: string = null;
+        manifestRootFolderPath = item.split('package.json')[0];
+        return new Promise((resolve, reject) => {
+            fs.readFile(manifestRootFolderPath+'target/npmlist.json', {encoding: 'utf-8'}, function(err, data) {
+                if(data){
+                    let packageListDependencies = JSON.parse(data);
+                    let packageDependencies = formatObj(packageListDependencies, ['name','version']);
+                    fs.writeFile(manifestRootFolderPath+'target/npmlist.json',JSON.stringify(packageDependencies), function(err) {
+                        if(err) {
+                            vscode.window.showErrorMessage(`Unable to format ${manifestRootFolderPath}target/npmlist.json`);
+                            reject(err);
+                        } else {
+                            let ePkgPath: any = manifestRootFolderPath+'target/npmlist.json';
```

Comment:
use https://nodejs.org/docs/latest/api/path.html#path_path_resolve_paths for path